version: '3.8'
services:
  localstack:
    image: localstack/localstack:1.2.0
    ports:
      - '4510-4559:4510-4559' # external service port range
      - '4566:4566'
    environment:
      - DEFAULT_REGION=eu-central-1 #TODO: to env
      - SERVICES=lambda,sqs,dynamodb,s3
      - DEBUG=1
      # TODO change to docker check network issue
      - LAMBDA_EXECUTOR=local
      # - LAMBDA_REMOTE_DOCKER=true
      # - LAMBDA_REMOVE_CONTAINERS=true
      - DATA_DIR=/tmp/localstack/data
      - DOCKER_HOST=unix:///var/run/docker.sock
    volumes:
      # - ./init-scripts:/docker-entrypoint-initaws.d
      - /var/run/docker.sock:/var/run/docker.sock
    networks:
      - crawler
  dynamodbAdmin:
    image: aaronshaf/dynamodb-admin:4.5.0
    ports: 
      - 8001:8001
    environment:
      DYNAMO_ENDPOINT: "http://localstack:4566"
      AWS_REGION: "eu-central-1" #TODO: to env
    depends_on:
      - localstack
    networks:
      - crawler
networks:
  crawler:
    external: false
    driver: bridge
    name: crawler